### ISSUE TITLE
Incomplete results have no message when a check omits `metadata.messages.incomplete`

### ISSUE THREAD

**Issue by chutchins25** [dequelabs/axe-core#5344]
## Description

When a check returns `undefined` (incomplete) and its JSON has no `metadata.messages.incomplete` entry, the node gets no message at all — `check.message` is `undefined` and `failureSummary` renders an empty bullet.

`lib/core/utils/publish-metadata.js` has two paths for incomplete messages and neither covers this case:

- The doT-template path is gated on `typeof messages.incomplete === 'object'`, which is `false` when the key is absent.
- The fallback then assigns `data.message = messages.incomplete`, i.e. `undefined`.

`incompleteFallbackMessage()` is imported in the file and used by `getDefaultMsg`, but `getDefaultMsg` is only reachable from inside `getIncompleteReason` — which the gate above stops from ever being called.

## Repro

`label-content-name-mismatch` has `pass` and `fail` messages but no `incomplete` message, and returns `undefined` for text that isn't human-interpretable:

```html
<button aria-label="comet">☄️</button>
```

Running that rule on `develop` (`540ecae9`):

```
incomplete count:    1
check.message:       undefined
node.failureSummary: "Fix any of the following:\n  "
```

## Expected

Fall back to `incompleteFallbackMessage()` so the node carries a usable message.

## Scope

Seven checks currently have a literal `return undefined` and no `incomplete` message: `label-content-name-mismatch`, `listitem`, `only-dlitems`, `only-listitems`, `important-letter-spacing`, `important-line-height`, `important-word-spacing`.

## Notes

Surfaced while reworking #5302, where a check-level `incomplete` message was added and then reverted as out of scope for that PR.

**Comment by chutchins25** [comment]
Fixed in #5367 (`4887c3a8`): `publishMetaData` now falls back to `incompleteFallbackMessage()` when a check declares no `metadata.messages.incomplete`, so those nodes carry a usable message instead of rendering an empty bullet.

@straker filed #5375 as the follow-up — stopping checks from returning a state they have no message for in the first place, by moving check tests onto `testutils.getCheckEvaluate`. The seven checks named here are the ones that issue covers.

Closing manually — the `Closes issue #5344` footer on #5367 didn't fire because the PR targeted `develop` rather than the default branch.
